lib/tasks/destroy_unused_providers.rb

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
+
+# Remove unused issuers
+#
+# When the IdP gets deployed, it generates a list of issuers that need to be removed and emails them
+# to us. These emails are available in Zendesk - check the "Suspended Tickets" bucket for the most
+# recent ticket titled:
+# "[production] identity-idp error: ServiceProviderSeeder::ExtraServiceProviderError".
+#
+# This script iterates over a list of those issuers, outputs the data that is to be deleted, and
+# then requires user confirmation before deleting the issuer and associated models.
+
+class DestroyUnusedProviders
+  attr_reader :destroy_list
+  def initialize(unused_issuers)
+    @destroy_list = unused_issuers.map { |issuer| DestroyableRecords.new(issuer) }
+  end
+
+  def run
+    @destroy_list.each do |records|
+      records.print_data
+
+      Rails.logger.debug do
+        "Type 'yes' and hit enter to continue and " \
+                         "destroy this service provider and associated records:\n"
+      end
+      continue = gets.chomp
```

Comment:
if we're using `gets` for interactive input, honestly I'd just use `STDOUT.puts` Rubocop might yell but it's probably fine.

To make testing easier (and also to trick rubocop) I would make the `stdin` and `stdout` IO's constructor parameters:

```ruby
def initialize(..., stdin: STDIN, stdout: STDOUT)
  @stdin = stdin
  @stdout = stdout
  # ...
end

# ....

continue = stdin.gets.chomp

stdout.puts 'aaa'
```
then in specs we pass in `StringIO.new`'s and we can manage them that way